src/compiler/utilities.ts

```diff
@@ -4573,6 +4573,18 @@ namespace ts {
         return hasSyntacticModifier(node, ModifierFlags.Static);
     }
 
+    export function hasOverrideModifier(node: Node): boolean {
+        return hasSyntacticModifier(node, ModifierFlags.Override);
+    }
+
+    export function hasAbstractModifier(node: Node): boolean {
+        return hasSyntacticModifier(node, ModifierFlags.Abstract);
+    }
+
+    export function hasAmbientModifier(node: Node): boolean {
+        return hasSyntacticModifier(node, ModifierFlags.Ambient);
+    }

```

Comment:
I'm finding the need for these somewhat surprising. How were we checking for ambient-ness and abtract-ness before? Are there cheaper ways?

Reply:
I'm not sure. Does it expensive? Even I founded cache inside `hasSyntacticModifier`

Reply:
I didn't see where this was used, so I don’t know for sure that this is what you’re looking for, but you can use `node.flags & NodeFlags.Ambient` to determine if any node is in an ambient context. It gets propagated down to all child nodes, so e.g. everything inside a `declare module` block will have that. Not sure if there’s an equivalent flag for abstract off the top of my head.